adrian-greksa/wk01/4-thurs/creditCardValidation/js/creditCardValidation.js

```diff
@@ -0,0 +1,76 @@
+// # Credit Card Validation
+
+// You're starting your own credit card business. You've come up with a new way to validate credit cards with a simple function called validateCreditCard that returns true or false.
+
+// Here are the rules for a valid number:
+
+// - Number must be 16 digits, all of them must be numbers
+// - You must have at least two different digits represented (all of the digits cannot be the same)
+// - The final digit must be even
+// - The sum of all the digits must be greater than 16
+
+// The following credit card numbers are valid:
+
+// - `9999-9999-8888-0000`
+// - `6666-6666-6666-1666`
+
+// The following credit card numbers are invalid:
+
+// - `a923-3211-9c01-1112` invalid characters
+// - `4444-4444-4444-4444` only one type of number
+// - `1111-1111-1111-1110` sum less than 16
+// - `6666-6666-6666-6661` odd final number
+
+// ## Example
+// ```
+// validateCreditCard('9999-9999-8888-0000'); // Returns: true
+// ```
+
+
+const validateCreditCard = function (creditCardNumber) {
+  let cardCharacters = creditCardNumber.replaceAll('-', ''); // .replaceAll(/-/g, '')
+  console.log(cardCharacters);
+
+  if (cardCharacters.length === 16 && cardCharacters.isInteger) { //check if 16 & are numbers
+    if (checkUnique()) {  //check if at least 2 unique characters
+      if (cardCharacters[15] % 2 === 0) {
+        if ()
+      } 
+    }
+  }    
+}
+
+const checkUnique = function() {
+  for (i = 0, i < 16; i++) {
+    let x = true;
+    if (cardCharacters[0] === cardCharacters[i]) {
+      x = false
+    } else {
+      x = false;
+    }
+  }
+}
```

Comment:
```suggestion
const checkUnique = function() {
    let unique = false;
    for (i = 0, i < number.length; i++) {
        if (cardCharacters[0] != cardCharacters[i]) {
            unique = true;
        }
    
    return unique;
}
```
-  With your **boolean variable**, you need to be careful about where it is declared (_inside or outside of the loop?_).
- I suggest when you have bugs related to loop logic that you walk through the loop one iteration at a time, carefully tracking the state of your variables.
- Once your loop is done, you need to **return** your variable, otherwise all the work done by the function will be for nothing!
- Try to avoid ambiguous variable names like `x` where possible.
- Try to favor using the **strict** not equal operator `!==`  over `!=`.

Also, rather than handling a boolean variable, the same effect could be achieved by **directly returning true or false** like this:
```
const checkUnique = function() {
    for (i = 0, i < number.length; i++) {
        if (cardCharacters[0] != cardCharacters[i]) {
            return true;
        }
    
    return false;
}
```